### PR TITLE
docs/tooltip-series-identification

### PR DESCRIPTION
**Pull request by madepiet**
Added information about tooltip for series options.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205846970443546

### CODE REVIEW

**Comment by bre1470** [comment]
The result looks like this: 
Skjermbilde 2024-01-11 kl  13 48 32

I am unsure about the recognition. I would probably not notice the remarks. Maybe if we have something in bold at the beginning, like `**Note:** ` could help; or maybe moving the paragraph up as the second one.

File: jsdoc/plugins/highcharts.jsdoc.js [hunk 1]
```diff
@@ -1109,15 +1109,26 @@ Move them up before functional code for JSDoc to see them.`.yellow
...
             var s = `
 
-In TypeScript the [type](series.${type}.type) option must always be set.
+**In TypeScript:**
+- the [type](series.${type}.type) option must always be set.
+- when accessing an array of series, the combined set of all series types is
+  represented by [Highcharts.SeriesOptionsType
+  ](/class-reference/Highcharts#.SeriesOptionsType).
+  It's recommended to cast to the specific type for better clarity and
+  correct access to all type-specific options.
```
Comment (bre1470) [inline jsdoc/plugins/highcharts.jsdoc.js:1127]: After providing the type property, TypeScript should narrow down to the series type. So no casting is needed. Just set/reset the type in every options object.

File: jsdoc/plugins/highcharts.jsdoc.js [hunk 2]
```diff
@@ -1143,6 +1154,11 @@ Highcharts.chart('container', {
     }]
 });
 \`\`\`
+
+When using the tooltip options with series, it is important to understand
```
Comment (bre1470) [inline jsdoc/plugins/highcharts.jsdoc.js:1158]: Can we have a `**Tooltip:** ` in front of it? And move TypeScript down here as I do not think that TypeScript should be the introduction text for series options.

Comment (KacperMadej) [reply jsdoc/plugins/highcharts.jsdoc.js:1158]: I forgot to remove this text section when testing via node_modules in the main HC repo. Sorry about that. My thinking is that tooltip is not that unique - a bigger excluded prop is data IMO, so a more general description is what I am aiming at. I like the idea of moving the TS section down here so it is not in the middle of the main description 👍

File: jsdoc/plugins/highcharts.jsdoc.js (hunk 1 above)
Comment (KacperMadej) [reply jsdoc/plugins/highcharts.jsdoc.js:1127]: Even with `type` set it fails when accessing via array index.

[Demo](https://stackblitz.com/edit/highcharts-angular-update-optimal-way-wyejng?file=src%2Fapp%2Fapp.component.ts,package.json).

image

Comment (bre1470) [reply jsdoc/plugins/highcharts.jsdoc.js]: Okay, I see what you mean. You should do a type check then:

```
if (chart.options.series?.[0].type === 'line') {
    chart.options.series[0].tooltip;
}
```

Comment (KacperMadej) [reply jsdoc/plugins/highcharts.jsdoc.js]: This is a way better way, thank you. Updated.

**Review by bre1470 — APPROVED** [review]
LGTM 👍

**Review by KacperMadej — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**